compile/api/src/main/scala/sbt/ClassToAPI.scala

```diff
@@ -127,14 +133,54 @@ object ClassToAPI {
   def upperBounds(ts: Array[Type]): api.Type =
     new api.Structure(lzy(types(ts)), lzyEmptyDefArray, lzyEmptyDefArray)
 
-  def fieldToDef(enclPkg: Option[String])(f: Field): api.FieldLike =
+  /** Parses the constant value represented by the given ConstantValue AttributeInfo. */
+  private def constantPoolConstantValue(cf: ClassFile, ai: classfile.AttributeInfo): AnyRef = {
+    assert(ai.name.exists(_ == "ConstantValue"), s"Non-ConstantValue attribute not supported: ${ai}")
+    import classfile.Constants._
+    cf.constantPool(classfile.Parser.entryIndex(ai)) match {
+      case classfile.Constant(ConstantString, nextOffset, _, _) =>
+        // follow the indirection from ConstantString to ConstantUTF8
+        val nextConstant = cf.constantPool(nextOffset)
+        nextConstant.value.getOrElse {
+          throw new RuntimeException(s"Empty UTF8 value in constant pool: $nextConstant")
```

Comment:
Ideally I'd like some other exception which extends runtime exception here.  I'm also clueless what we should actually do if we encounter this.  Is it safe to ignore?  Does it mean a corrupted classfile?  

Ideally I'd like a string telling people what to do if we understand why it would happen.